contrib/pkg/deprovision/awstagdeprovision.go

```diff
@@ -120,6 +133,66 @@ func completeAWSUninstaller(o *aws.ClusterUninstaller, logLevel string, args []s
 	return nil
 }
 
+func (o *awsTagDeprovisionOpts) run() error {
+	return wait.ExponentialBackoff(
+		// Start the backoff at 5 minutes, double it each time, to a cap of 24 hours.
+		wait.Backoff{
+			Duration: 5 * time.Minute,
+			Factor:   2,
+			Steps:    1 << 8, // large enough to make cap the effective bound
+			Cap:      24 * time.Hour,
+		},
+		func() (done bool, returnErr error) {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+			defer cancel()
+			uninstaller := &aws.ClusterUninstaller{
+				Filters: o.filters,
+				Logger:  o.logger,
+				Region:  o.region,
+			}
+			blockedResources, err := uninstaller.RunWithContext(ctx)
+			if len(blockedResources) == 0 {
+				return true, err
+			}
+			if o.clusterDeprovision == "" {
+				return
+			}
+			kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+				clientcmd.NewDefaultClientConfigLoadingRules(),
+				&clientcmd.ConfigOverrides{},
+			)
+			namespace, _, err := kubeconfig.Namespace()
+			if err != nil {
+				o.logger.WithError(err).Error("could not get the namespace")
+				return
+			}
```

Comment:
does this use the context setup using the kubeconfig to get the namespace? How many users would use something like that? _I personally haven't_ So should we also allow for setting namespace directly?